src/pass/llvmgen.cpp

```diff
@@ -392,6 +390,63 @@ Value* LLVMGen::visit(Store& store)
     return CreateStore(val, addr);
 }
 
+Value* LLVMGen::visit(AtomicOp& e)
+{
+    auto addr = eval(e.addr);
+    auto val = eval(e.val);
+    llvm::AtomicRMWInst::BinOp instr_type;
+    switch (e.op) {
+        case MathOp::ADD: {
+            if (e.addr->type.is_float()) {
```

Comment:
I'm pretty sure this doesn't work. because e.addr is of a ptr type. you either check against the val->type. or against, e.addr->type.deref()